flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -659,14 +661,36 @@ private CompletableFuture<CleanupJobState> handleJobManagerRunnerResult(
         return jobReachedTerminalState(jobManagerRunnerResult.getExecutionGraphInfo());
     }
 
-    enum CleanupJobState {
-        LOCAL,
-        GLOBAL
+    private static class CleanupJobStateWithJobStatusIfGlobal {
+
+        private static final CleanupJobStateWithJobStatusIfGlobal LOCAL =
+                new CleanupJobStateWithJobStatusIfGlobal(null);
```

Comment:
Do we really need a nullable jobstatus in the first place?

* jobManagerRunnerFailed could use initializing
* jobReachedTerminalState could use suspended

Reply:
That's a good point. That keeps it consistent. Thanks for questioning that approach... 👍